src/components/typography/headers/Header.tsx

```diff
@@ -7,8 +7,8 @@ import useUITheme from '../../theming/useUITheme';
 export type HeaderProps = {
   /** Underlying HTML element tag to use. */
   size: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
-  /** The text of the header. */
-  text: string;
+  /** The text of the header. This must either be a string or a <span> element. */
+  text: string | ReactElement<HTMLSpanElement>;
```

Comment:
So the `<HTMLSpanElement>` is a hint for the developer but isn't enforced by TypeScript?  (As you note below.)

Reply:
@eikonomega and I spoke briefly on slack about this. I think the plan is to make this:

```suggestion
  text: ReactNode;
```

It's a little more flexible for unforeseen scenarios.